app/models/claim/base_claim.rb

```diff
@@ -321,10 +321,8 @@ def representation_orders
 
     def earliest_representation_order
       return if defendants.empty?
-      defendants
-        .map(&:earliest_representation_order)
-        .compact
-        .sort_by(&:representation_order_date).first
+
+      defendants.filter_map(&:earliest_representation_order).min_by(&:representation_order_date)
```

Comment:
and `min_by`, but i trust this works for dates!

https://ruby-doc.org/core-2.7.3/Enumerable.html#method-i-min_by

Reply:
certainly seems to
```
arr = [Date.today - 1, Date.today - 2, Date.today]
arr.min_by{|x| x.to_date }
```